.github/workflows/ci-workflow.yml

```diff
@@ -0,0 +1,74 @@
+name: native-metrics CI
+
+on:
+  push:
+    branches:
+    - open-source-main
+  pull_request:
+    branches:
+    - '**'
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        node-version: [14.x]
+
+    steps:
+    - uses: actions/checkout@v2
+    - name: Use Node.js ${{ matrix.node-version }}
+      uses: actions/setup-node@v1
+      with:
+        node-version: ${{ matrix.node-version }}
+    - name: Install Dependencies
+      run: npm ci
+    - name: Rebuild
+      run: npm run rebuild
+    - name: Run Linting
+      run: npm run lint
+
+  linux:
```

Comment:
while i almost lead you astray, and maybe we can solve a different way eventually, i think this makes sense as we wan to run the suite against the same built artifacts.